tensorboard/plugins/hparams/summary.py

```diff
@@ -0,0 +1,147 @@
+
+  Returns:
+    A summary protobuffer containing the experiment definition.
+  """
+  experiment = api_pb2.Experiment(
+      description=description,
+      user=user,
+      time_created=timestamp_pb2.Timestamp().FromDatetime(time_created),
+      hparam_infos=hparam_infos,
+      metric_infos=metric_infos)
+  return _summary(metadata.EXPERIMENT_TAG,
+                  data_oneof_field="experiment",
+                  protobuffer=experiment)
+
+
+def session_start_pb(hparams,
+                     checkpoint_uri="",
+                     monitor_url="",
+                     group_name=""):
+  """Creates a summary that contains a training session metadata information.
+  One such summary per training session should be created. Each should have
+  a different run.
+
+  Arguments:
+    hparams: A dictionary with string keys. Describes the hyperparameter values
+             used in the session mappng each hyperparameter name to its value.
+             Supported value types are  bool, int, float, or str.
+    checkpoint_uri: See the comment for the field with the same name of
+                    plugin_data_pb2.SessionStartInfo.
+    monitor_url: See the comment for the field with the same name of
+                 plugin_data_pb2.SessionStartInfo.
+    group_name:  See the comment for the field with the same name of
+                 plugin_data_pb2.SessionStartInfo.
+  Returns:
+    Returns the summary protobuffer mentioned above.
+  """
+  supported_types={float:None, str:None, int:None, bool:None}
+  session_start_info = plugin_data_pb2.SessionStartInfo(
+      checkpoint_uri=checkpoint_uri,
+      monitor_url=monitor_url,
+      group_name=group_name)
+  for (hp_name, hp_val) in six.iteritems(hparams):
+    assert type(hp_val) in supported_types
+    if type(hp_val) is float or type(hp_val) is int:
+      session_start_info.hparams[hp_name].number_value = hp_val
+    elif type(hp_val) is str:
+      session_start_info.hparams[hp_name].string_value = hp_val
+    else: # type(hp_val) is bool:
+      session_start_info.hparams[hp_name].bool_value = hp_val
+  return _summary(metadata.SESSION_START_INFO_TAG,
+                  data_oneof_field="session_start_info",
+                  protobuffer=session_start_info)
+
+
+def session_end_pb(status):
+  """Creates a summary that contains status information for a completed
+  training session. Should be exported after the training session is completed.
+  One such summary per training session should be created. Each should have
+  a different run.
+  Arguments:
+    status: A tensorboard.hparams.Status enumeration value denoting the
+            status of the session.
+
+  Returns:
+    Returns the summary protobuffer mentioned above.
+  """
+  return _summary(metadata.SESSION_END_INFO_TAG,
+                  data_oneof_field="session_end_info",
+                  protobuffer=plugin_data_pb2.SessionEndInfo(status=status))
+
+
+def _summary(tag, data_oneof_field, protobuffer):
```

Comment:
Passing around a field name for a later call to getattr() is a little hacky and I'm not sure it actually saves much code compared to just writing out the HParamsPluginData directly and passing it through _summary() and create_summary_metadata().

E.g. instead of the code above for session_end_info, you could just do

```py
plugin_data = plugin_data_pb2.HParamsPluginData()
plugin_data.session_end_info.status = status
return _summary(metadata.SESSION_END_INFO_TAG, plugin_data)
```

I think this is actually roughly the same amount of boilerplate as what exists in the code right now, but it's a bit more type-safe since it doesn't use a stringified field name, and it allows the helpers to be even simpler than they are now.

Reply:
+1 for this suggestion.